### PR TITLE
Allow user to hide chat messages on a livestream

### PR DESCRIPTION
**Pull request by kekkyojin**
## PR Checklist

Please check all that apply to this PR using "x":

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [x] I have checked that this PR does not introduce a breaking change

## PR Type

What kind of change does this PR introduce?

- [x] Feature

## Fixes

Issue Number: #261 

## What is the current behavior?
Chat is always shown
## What is the new behavior?
Chat is hidden by default and user is able to show it or hides it. The setting is kept until user changes it
## Other information
In order to show again the chat, user must click on the gear icon and choose the option. When chat messages are displayed, only a close button appears.

As more options are implemented -like show timestamps-, gear related code will need to be changed and string for the “Show chat messages” item be updated as needed.

### CODE REVIEW

**Comment by kekkyojin** [comment]
Ping @akinwale

**Comment by akinwale** [comment]
@kekkyojin I posted a question about some bits of code. Did you get a chance to look at it?

File: app/src/main/java/com/odysee/app/adapter/ChatMessageListAdapter.java [hunk 1]
```diff
@@ -72,33 +77,48 @@ public ViewHolder(View v) {
...
+                    streamerClaimId,
+                    message.getHandler(),
+                    vh.textMessage,
+                    context));
+        } else {
+            vh.itemView.setVisibility(View.GONE);
+            lp.height = 0;
+            lp.width = 0;
+            vh.itemView.setLayoutParams(lp);
+        }
```
[the remark this answers is not in the document]
Comment (kekkyojin) [reply app/src/main/java/com/odysee/app/adapter/ChatMessageListAdapter.java:122]: This way state is kept on the adapter.

**Review by akinwale — APPROVED** [review]
Looks good. Thanks.

**Comment by tzarebczan** [comment]
Let's hide by default and then we can open an issue to add a setting for this.

**Pull request merged**